mrmustard/lab_dev/transformations/base.py

```diff
@@ -107,7 +151,28 @@ def from_bargmann(
         c = math.astensor(triple[2])
         shape_check(A, b, 2 * len(modes), "Bargmann")
         s = set(modes)
-        return Unitary._from_attributes(name, Bargmann(A, b, c), Wires(set(), set(), s, s))
+        return Unitary._from_attributes(
+            representation=Bargmann(A, b, c), wires=Wires(set(), set(), s, s), name=name
+        )
+
+
+class Map(Transformation):
+    r"""A CircuitComponent with input and output wires and same modes on bra and ket sides.
```

Comment:
I am confused. If modes have to be the same, why do you specify modes_out and modes_in? Please reword.

Reply:
Oh, it's the old docstring (when Channel had only `modes`), I'll go through all the docstrings in the next pr (#406)